src/hooks/useEventListener.jsx

```diff
@@ -0,0 +1,24 @@
+import { useRef, useEffect } from 'react';
+
+function useEventListener(eventName, handler, element = document) {
+  const savedHandler = useRef();
+
+  useEffect(() => {
+    savedHandler.current = handler;
+  }, [ handler ]);
+
+  useEffect(() => {
+    const isSupported = element && element.addEventListener;
+
+    if (!isSupported) return undefined;
+
+    const eventListener = (event) => savedHandler.current(event);
+    element.addEventListener(eventName, eventListener);
```

Comment:
You can just wrap this in an if - 
```suggestion
    if (isSupported) {
      element.addEventListener(eventName, eventListener);
    }
```

Then you do not need to have an if to break out of the logic